### ISSUE TITLE
cannot find io/realm/XX

### ISSUE THREAD

**Issue by JarvanMo** [realm/realm-java#2504]
Error:Execution failed for task ':GsouDomestic:transformClassesWithRealmTransformerForDebug'.
> cannot find io/realm/PlayListForRealmEntityRealmProxy: io.realm.PlayListForRealmEntityRealmProxy found in io/realm/PlayListForRealmEntityRealmProxy.class

I can run my module on my PC, but my friend can't run it .What's wrong?

**Comment by beeender** [comment]
Could you please share the gradle file? Is the Realm used in a Android library project?

**Comment by JarvanMo** [comment]
Yes, I have two libraries. My os is ubuntu ,and my friend's os is Windows 10.
Here is my application project's  build.gradle:

apply plugin: 'com.android.application'
apply plugin: 'realm-android'



android {

    compileSdkVersion 23
    buildToolsVersion '23.0.2'

    defaultConfig {
        applicationId "co.tton.gsoudomestic"
        minSdkVersion 17
        targetSdkVersion 23
        versionCode 1
        versionName "1.0"
        multiDexEnabled true
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }


    packagingOptions {
        exclude 'META-INF/DEPENDENCIES.txt'
        exclude 'META-INF/DEPENDENCIES'
        exclude 'META-INF/dependencies.txt'
        exclude 'META-INF/LICENSE.txt'
        exclude 'META-INF/LICENSE'
        exclude 'META-INF/license.txt'
        exclude 'META-INF/LGPL2.1'
        exclude 'META-INF/NOTICE.txt'
        exclude 'META-INF/NOTICE'
        exclude 'META-INF/notice.txt'
        exclude 'META-INF/services/javax.annotation.processing.Processor' // butterknife
        exclude "lib/arm64-v8a/librealm-jni.so"
        exclude "lib/armeabi-v7a/librealm-jni.so"
    }

    lintOptions {
        disable "ResourceType"
    }
}


dependencies {
    compile fileTree(include: ['*.jar'], dir: 'libs')
    compile project(':GsouBase')
}
buildscript {


    repositories {
        jcenter()
    }


    dependencies {
        classpath "io.realm:realm-gradle-plugin:0.88.2"
    }
}


Here is one of my  library project's  build.gradle:

apply plugin: 'com.android.library'
apply plugin: 'realm-android'

android {


    compileSdkVersion 23
    buildToolsVersion "23.0.2"

    defaultConfig {
        minSdkVersion 17
        targetSdkVersion 23
        versionCode 1
        versionName "1.0"
        multiDexEnabled true
    }


    packagingOptions {
        exclude 'META-INF/DEPENDENCIES.txt'
        exclude 'META-INF/DEPENDENCIES'
        exclude 'META-INF/dependencies.txt'
        exclude 'META-INF/LICENSE.txt'
        exclude 'META-INF/LICENSE'
        exclude 'META-INF/license.txt'
        exclude 'META-INF/LGPL2.1'
        exclude 'META-INF/NOTICE.txt'
        exclude 'META-INF/NOTICE'
        exclude 'META-INF/notice.txt'
        exclude 'META-INF/services/javax.annotation.processing.Processor' // butterknife
        exclude "lib/arm64-v8a/librealm-jni.so"
        exclude "lib/armeabi-v7a/librealm-jni.so"
    }

    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
}


dependencies {
    compile fileTree(include: ['*.jar'], dir: 'libs')
    testCompile 'junit:junit:4.12'
    compile project(':MLibrary')
//    compile 'io.realm:realm-android:0.87.5'
    //    compile 'com.specyci:residemenu:1.6+'
    compile project(':SlidingMenuLibrary')
    compile 'com.android.support:multidex:1.0.1'
    compile project(':CyberLink4Android')
    compile project(':croplibrary')
}

buildscript {


    repositories {
        jcenter()
    }


    dependencies {
        classpath "io.realm:realm-gradle-plugin:0.88.2"
    }
}


Here's the other one:
apply plugin: 'com.android.library'
apply plugin: 'realm-android'

android {


    compileSdkVersion 23
    buildToolsVersion "23.0.2"

    defaultConfig {
        minSdkVersion 17
        targetSdkVersion 23
        versionCode 1
        versionName "1.0"
        multiDexEnabled true
    }


    packagingOptions {
        exclude 'META-INF/DEPENDENCIES.txt'
        exclude 'META-INF/DEPENDENCIES'
        exclude 'META-INF/dependencies.txt'
        exclude 'META-INF/LICENSE.txt'
        exclude 'META-INF/LICENSE'
        exclude 'META-INF/license.txt'
        exclude 'META-INF/LGPL2.1'
        exclude 'META-INF/NOTICE.txt'
        exclude 'META-INF/NOTICE'
        exclude 'META-INF/notice.txt'
        exclude 'META-INF/services/javax.annotation.processing.Processor' // butterknife
        exclude "lib/arm64-v8a/librealm-jni.so"
        exclude "lib/armeabi-v7a/librealm-jni.so"
    }

    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
}


dependencies {
    compile fileTree(include: ['*.jar'], dir: 'libs')
    testCompile 'junit:junit:4.12'
    compile project(':MLibrary')
//    compile 'io.realm:realm-android:0.87.5'
    //    compile 'com.specyci:residemenu:1.6+'
    compile project(':SlidingMenuLibrary')
    compile 'com.android.support:multidex:1.0.1'
    compile project(':CyberLink4Android')
    compile project(':croplibrary')
}

buildscript {


    repositories {
        jcenter()
    }


    dependencies {
        classpath "io.realm:realm-gradle-plugin:0.88.2"
    }
}

**Comment by emanuelez** [comment]
Did you try a project clean on your friends computer?

**Comment by JarvanMo** [comment]
I had tried many times .Actually I cannot run it on my windows neither, but it works well on Linux.

**Comment by emanuelez** [comment]
I tried building a sample project on a windows machine and I could not reproduce the issue. Would it be possible for you to share your project (or a smaller sample project that shows the same issue) with us, even in a private way?

**Comment by JarvanMo** [comment]
I'll send you an email

**Comment by emanuelez** [comment]
Thank you, I received it. We will take a look at it.

**Comment by beeender** [comment]
@JarvanMo Just tried your project, it doesn't compile on linux either.

First thing is `GsouBase` is using Realm in a library project, it needs to define a `RealmModule` as described in https://realm.io/docs/java/latest/#schemas. Otherwise it will complain about the duplicate dex class "RealmDefaultModule`.

After fix that, the build failed with 
```
Execution failed for task ':TestForRealm:prepareDebugUnitTestDependencies'.
 Dependency Error. See console for details.
```

It doesn't seem to be related with Realm, maybe you could look into it so we can go further?

**Comment by JarvanMo** [comment]
Ok,I'll have a look into it.

**Comment by CarlosMChica** [comment]
Hello, I've run into the same issue.

In my case, I'm able to run the build just fine on mac but it does not work on windows (tried in two different machines with the same result). I'm using realm in a library project too.

The only difference that I can see with the original issue is that the app builds just find for production (I can install and run the app from windows as well), but if fails when running the test task. 

The Gradle task that I'm running is 'check connectedCheck' and the output is as follows:

 `Build failure reason Execution failed for task ':persistence:transformClassesWithRealmTransformerForDebugAndroidTest'. java.lang.RuntimeException: cannot find io/realm/XXXRealmProxy: io.realm.XXXRealmProxy found in io/realm/XXXRealmProxy.class`

I guess it has nothing to do but it seems weird that it works on Linux (as the original issue says) and mac but not on windows. I don't really know what's the issue here but I noticed that the output states it's trying to find the class using "/" for directories instead of "\" on windows?

**Comment by beeender** [comment]
@CarlosMChica Yeah, i think you figured out what the problem is.
The fix is ready #2540 . And since this is the dup error to #2484, I am just closing this now. Let's discuss in #2484 for this windows build issue.
